File: 프로그래머스/Lv.0/181858. 무작위로 K개의 수 뽑기/무작위로 K개의 수 뽑기.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr, int k) {
        List<Integer> answer = new ArrayList<Integer>();       
        boolean[] nums = new boolean[100001];
        for(int i=0;i<arr.length;i++){
            if(nums[arr[i]]) continue;
            nums[arr[i]] = true;
            answer.add(arr[i]);
            k--;
            if(k==0) break;
        }
        for(int i=0;i<k;i++)
            answer.add(-1);
        return answer.stream().mapToInt(i->i).toArray();
    }
}

File: 프로그래머스/Lv.0/181858. 무작위로 K개의 수 뽑기/무작위로 K개의 수 뽑기.py
def solution(arr, k):
    nums = [False for _ in range(100001)]
    answer = []
    for a in arr:
        if nums[a]:
            continue
        nums[a] = True
        answer.append(a)
        k-=1
        if k==0:
            break
    answer.extend([-1] * k)
    return answer